crux-mir/src/Mir/Language.hs

```diff
@@ -155,30 +155,33 @@ withMirLogging computation =
         ?injectMirLogMessage = LoggingMir
      in computation
 
-mainWithOutputConfig :: (Maybe Crux.OutputOptions -> OutputConfig MirLogging)
-                     -> BindExtraOverridesFn -> IO ExitCode
-mainWithOutputConfig mkOutCfg bindExtra =
+mainWithOutputConfig ::
+    (forall sym. IO (s sym)) ->
+    (Maybe Crux.OutputOptions -> OutputConfig MirLogging) ->
+    BindExtraOverridesFn s -> IO ExitCode
+mainWithOutputConfig initS mkOutCfg bindExtra =
     withMirLogging $
     Crux.loadOptions mkOutCfg "crux-mir" Paths_crux_mir.version mirConfig
-        $ runTestsWithExtraOverrides bindExtra
+        $ runTestsWithExtraOverrides initS bindExtra
 
-type BindExtraOverridesFn = forall sym bak p t st fs args ret blocks rtp a r.
+type BindExtraOverridesFn s = forall sym bak p t st fs args ret blocks rtp a r.
```

Comment:
It would be worth some Haddocks saying what this `s` type parameter is, along with an explanation of why it is universally quantified (unlike all of the other type variables).